wordcloud.py

```diff
@@ -57,41 +55,41 @@ def process_string(text_to_process):
 
 def determine_occurrences(list_data):
     """Add list to dictionary with value based on occurrence."""
-    word_to_occurrence = {}
+    word_to_count = {}
     for word in list_data:
-        word_to_occurrence[word] = list_data.count(word)
-    return word_to_occurrence
+        word_to_count[word] = list_data.count(word)
```

Comment:
If the words are `["this", "string", "string"]`
The result should be: `{"this": 1, "string": 2}`.
What are the actual results instead?

Reply:
I've tested it and the results look like that to me.  Can you show me what you get please?

Reply:
Apologies, you're right. The results are correct. I somehow thought that doing `dictionary[key] = value` multiple times created duplicates, but using the same key just overrides the previous value. 🙄 That does reveal that there is an inefficiency in the code.

Consider the following code...
```
word_to_count = {}
words = ["echo", "echo", "echo"]
for word in words:
    word_to_count[word] = words.count(word)
```
How many times will words.count() be run for the same word? 3 times right?

1. `word_to_count["echo"] = words.count("echo")`
2. `word_to_count["echo"] = words.count("echo")`
3. `word_to_count["echo"] = words.count("echo")`

See some repetition here? 😉 
1 adds the value 3 to word_to_count with the key "echo".
2 replaces the value of the existing dictionary entry with key "echo" with the value 3.
3 does the same as 2.
So 1 and 2 are a waste because they are replaced by 3.

It would be better if we only counted "echo" once. *

There are a number of different ways to do this. One simple way though would be to convert the word list to a set. Doing so removes all duplicates (as you remember sets cannot contain duplicates so converting a list to a set automatically removes them). 😄 

Simply do:
```
...
for word in set(words):
    ...
```

\* The code works, so why fix the inefficiency? I think it's a good habit to think about efficiency when writing code, but it's also easy to overthink a problem and try to make it more efficient, and therefore more complex, when it doesn't need to be ("pre-mature optimization is the root of all evil" is a saying in IT 😛). This could very well be a case where the inefficiency is small and doesn't matter. However, I think it's helpful to be aware of inefficiencies in our code, even if we decide not fix them to keep the code simple.

Reply:
Great idea!  I have implemented it, cautiously.  Let me know if you think I need to remove it.